src/services/DexPoolScanner.ts

```diff
@@ -0,0 +1,180 @@
+import {injectable} from 'inversify';
+import {Logger} from 'winston';
+import {StaticJsonRpcProvider} from '@ethersproject/providers';
+
+import {ChainsIds} from '../types/ChainsIds.js';
+import {DexProtocolName} from '../types/DexProtocolName.js';
+import {MappingRepository} from '../repositories/MappingRepository.js';
+import {BlockchainProviderRepository} from '../repositories/BlockchainProviderRepository.js';
+import {DexProtocolFactory, PoolCreatedEvent} from '../factories/DexProtocolFactory.js';
+import {DexProtocolInterface} from '../interfaces/DexProtocolInterface.js';
+import {DexRepository} from '../repositories/DexRepository.js';
+import Application from '../lib/Application.js';
+import Settings from '../types/Settings.js';
+
+export type Result = {
+  fromBlock: number;
+  toBlock?: number;
+  success: boolean;
+  synchronized?: boolean;
+  waitBlockRange?: boolean;
+};
+
+export type UniswapPoolCreatedEvent = {
+  token0: string;
+  token1: string;
+  fee: bigint;
+  pool: string;
+  anchor: number;
+};
+
+@injectable()
+export class DexPoolScanner {
+  logger: Logger;
+  settings: Settings;
+  mappingRepository!: MappingRepository;
+  blockchainProviderRepository!: BlockchainProviderRepository;
+  dexRepository!: DexRepository;
+
+  startBlock: number;
+  dexProtocolName!: DexProtocolName;
+  step: number;
+  provider: StaticJsonRpcProvider | undefined;
+  chainId!: ChainsIds;
+  dexProtocol: DexProtocolInterface;
+  logPrefix: string;
+
+  constructor(chainId: ChainsIds, dexProtocolName: DexProtocolName) {
```

Comment:
I'm not sure if injectable class will work with constructor as expected. for ode dex probably it will, but not for multiple. chainID should be pass to run/apply method. because this is parameter. we not constructing new instance for every dex. same instance should work for any dex.